com.unity.render-pipelines.high-definition/Editor/Lighting/HDLightUI.PresetInspector.cs

```diff
@@ -0,0 +1,30 @@
+using System;
+using System.Linq.Expressions;
+using System.Reflection;
+using UnityEngine;
+using UnityEngine.Rendering.HighDefinition;
+using UnityEngine.Rendering;
+
+namespace UnityEditor.Rendering.HighDefinition
+{
+    using CED = CoreEditorDrawer<SerializedHDLight>;
+
+    static class HDLightUIPreset
+    {
+        [HDRPHelpURL("Light-Component")]
+        enum Expandable
+        {
+            General = 1 << 0,
+            Emission = 1 << 2,
+        }
+
+        static readonly ExpandedState<Expandable, Light> k_ExpandedState = new ExpandedState<Expandable, Light>(~(-1), "HDRP-preset");
```

Comment:
```suggestion
        static readonly ExpandedState<Expandable, Light> k_ExpandedState = new ExpandedState<Expandable, Light>(~(-1), "HDRP:LightUIPreset");
```

Overlaping key with different sections on : `HDCameraUIPreset`

Reply:
if that's the case then should I also update for unique strings HDProbeUI, LocalVolumetricFogUI, HDRenderPipelineUI etc?
For ex. i see we have :
```
static readonly ExpandedState<Expandable, HDRenderPipelineAsset> k_ExpandedState = new(Expandable.Rendering, "HDRP");
static readonly ExpandedState<ExpandableShadows, HDRenderPipelineAsset> k_LightsExpandedState = new(0, "HDRP:Shadows")
```
 but also
```
readonly static ExpandedState<Expandable, ProbeVolume> k_ExpandedStateVolume = new ExpandedState<Expandable, ProbeVolume>(Expandable.Volume, "HDRP");
readonly static ExpandedState<Expandable, ProbeVolume> k_ExpandedStateProbes = new ExpandedState<Expandable, ProbeVolume>(Expandable.Probes, "HDRP");
```
@RSlysz thoughts?

Reply:
Ok, I saw the implementation and it is doing: `string key = $"{prefix}:{typeof(TTarget).Name}:{typeof(TState).Name}:UI_State";` so they won't be overlaped. Then you do not need the - preset as it will get the name of the class HDLightUIPreset:Expanded and HDCameraUIPreset:Expanded.

Reply:
I do not know why but I got confused :) to much time from the game jam maybe xD

Reply:
Well, If I remember well this string is a prefix of the key which is computed this way:
`<Prefix>:<TypeName>:<StateName>:UI_State`
Prefix: to distinguish amongst SRP (usually HDRP / URP / CoreRP)
TypeName: the type name of the component, second template argument (here "Light")
StateName: the type name of the enum encoding the state, first template argument argument ("here "Expandable")

So yes, using ':' is a better separator than '-'. But at the same time it is still part of the prefix disambiguation. Just here we do not only disambiguish by SRP but also by inspector kind (preset or not).
So for me "HDRP-Preset" works fine.
We can chose to use the separator, adding a new segment in the key but in this case, "Light" and "UI" should not be in it as already in the kay name. So it would be "HDRP:Preset"

This is temporary anyway so both works for me (but try not to duplicate information please, the key is already long)

Reply:
(the key used to store the EditorPref value)

Reply:
"HDRP:Shadows" for instance is odd as Shadow is already induced but the enum used "ExpandableShadows". IMO it should have remained 
`static readonly ExpandedState<ExpandableShadows, HDRenderPipelineAsset> k_LightsExpandedState = new(0, "HDRP")
`

Reply:
yeah, we should remove the :Shadows:

Reply:
For the ProbeVolume it is ill constructed and conflict o_o It should not

Reply:
this should be fixed in the next commit then :)

Reply:
Sorry Alex, just saw your message. Git don't refresh itself :D